plugins/draco/io/DracoReader.cpp

```diff
@@ -0,0 +1,363 @@
+        case 1:
+            got_data = attr->ConvertValue<T, 1>(val_index,
+                                                data.data() + i.value() * num_components);
+            break;
+        case 2:
+            got_data = attr->ConvertValue<T, 2>(val_index,
+                                                data.data() + i.value() * num_components);
+            break;
+        case 3:
+            got_data = attr->ConvertValue<T, 3>(val_index,
+                                                data.data() + i.value() * num_components);
+            break;
+        case 4:
+            got_data = attr->ConvertValue<T, 4>(val_index,
+                                                data.data() + i.value() * num_components);
+            break;
+        default:
+        break;
+    }
+    if (!got_data)
+       throw pdal_error("CopyAttributeData unable to read data for point ");
+    }
+
+    return true;
+}
+
+Dimension::Type getPdalType(draco::DataType t)
+{
+    using namespace draco;
+    switch (t)
+    {
+        case DT_INT8:
+            return Dimension::Type::Signed8;
+        case DT_UINT8:
+            return Dimension::Type::Unsigned8;
+        case DT_INT16:
+            return Dimension::Type::Signed16;
+        case DT_UINT16:
+            return Dimension::Type::Unsigned16;
+        case DT_INT32:
+            return Dimension::Type::Signed32;
+        case DT_UINT32:
+            return Dimension::Type::Unsigned32;
+        case DT_INT64:
+            return Dimension::Type::Signed64;
+        case DT_UINT64:
+            return Dimension::Type::Unsigned64;
+        case DT_FLOAT32:
+            return Dimension::Type::Float;
+        case DT_FLOAT64:
+            return Dimension::Type::Double;
+
+        default:
+            // Not supported draco domain types
+            throw pdal_error("Invalid Dim type from Draco");
+    }
+}
+
+void DracoReader::addArgs(ProgramArgs& args)
+{
+}
+
+void DracoReader::prepared(PointTableRef table)
+{
+    if (m_filename.empty())
+        throwError("Required argument 'filename' (Draco array name) "
+            "not provided.");
+}
+
+
+void DracoReader::initialize()
+{
+    if (!FileUtils::fileExists(m_filename))
+        throwError("File '" + m_filename + "' does not exist");
+
+    m_istreamPtr = Utils::openFile(m_filename, true);
+    if (!m_istreamPtr)
+        throwError("Unable to open file '" + m_filename + "' ");
+
+    m_data.assign(std::istreambuf_iterator<char>(*m_istreamPtr),
+                std::istreambuf_iterator<char>());
+    Utils::closeFile(m_istreamPtr);
+
+    m_draco_buffer.Init(m_data.data(), m_data.size());
+
+    draco::Decoder decoder;
+    auto geom_status = draco::Decoder::GetEncodedGeometryType(&m_draco_buffer);
+    if (!geom_status.ok())
+    {
+        return throwError(geom_status.status().error_msg());
+    }
+
+    const draco::EncodedGeometryType geom_type = geom_status.value();
+
+    draco::Decoder::GetEncodedGeometryType(&m_draco_buffer);
```

Comment:
Does this method provide some kind of error status into `m_draco_buffer` or somewhere else to check the result of this operation?